micrometer-core/src/main/java/io/micrometer/core/instrument/binder/hystrix/MicrometerMetricsPublisherThreadPool.java

```diff
@@ -0,0 +1,142 @@
+ */
+package io.micrometer.core.instrument.binder.hystrix;
+
+import com.netflix.hystrix.HystrixThreadPoolKey;
+import com.netflix.hystrix.HystrixThreadPoolMetrics;
+import com.netflix.hystrix.HystrixThreadPoolProperties;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisherThreadPool;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+/**
+ * Micrometer publisher for Hystrix thread pool metrics.
+ */
+@NonNullApi
+@NonNullFields
+public class MicrometerMetricsPublisherThreadPool implements HystrixMetricsPublisherThreadPool {
+  private static final String NAME_HYSTRIX_THREADPOOL = "hystrix.threadpool";
+
+  private final MeterRegistry meterRegistry;
+  private final HystrixThreadPoolKey threadPoolKey;
+  private final HystrixThreadPoolMetrics metrics;
+  private final HystrixThreadPoolProperties properties;
+  private final HystrixMetricsPublisherThreadPool metricsPublisherForThreadPool;
+  private final Tags tags;
+
+  public MicrometerMetricsPublisherThreadPool(
+      final MeterRegistry meterRegistry,
+      final HystrixThreadPoolKey threadPoolKey,
+      final HystrixThreadPoolMetrics metrics,
+      final HystrixThreadPoolProperties properties,
+      final HystrixMetricsPublisherThreadPool metricsPublisherForThreadPool) {
+    this.meterRegistry = meterRegistry;
+    this.threadPoolKey = threadPoolKey;
+    this.metrics = metrics;
+    this.properties = properties;
+    this.metricsPublisherForThreadPool = metricsPublisherForThreadPool;
+
+    this.tags = Tags.of("key", threadPoolKey.name());
+  }
+
+  @Override
+  public void initialize() {
+    metricsPublisherForThreadPool.initialize();
+
+    Gauge.builder(metricName("threads.active.current.count"), metrics::getCurrentActiveCount)
+        .description("The approximate number of threads that are actively executing tasks.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.active.rolling.max"), metrics::getRollingMaxActiveThreads)
+        .description("Rolling max number of active threads during rolling statistical window.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.executed.cumulative.count"), metrics::getCumulativeCountThreadsExecuted)
+        .description("Cumulative count of number of threads executed since the start of the application.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.rejected.cumulative.count"), metrics::getCumulativeCountThreadsRejected)
+        .description("Cumulative count of number of threads rejected since the start of the application.")
+        .tags(tags)
+        .register(meterRegistry);
+
+    Gauge.builder(metricName("threads.executed.rolling.count"), metrics::getRollingCountThreadsExecuted)
```

Comment:
I'm hesitant to include the rolling counts since that information is exposed via the the cumulative ones. Are you comfortable with leaving them out?

Reply:
Done.